torch/cuda/_utils.py

```diff
@@ -24,12 +22,9 @@ def _get_device_index(device: Union[Device, str, int], optional: bool = False,
     device_idx: Optional[int]
     if isinstance(device, torch.device):
         dev_type = device.type
-        if allow_cpu:
-            if device.type not in {'cuda', 'cpu'}:
-                raise ValueError('Expected a cuda or cpu device, but got: {}'.format(device))
-        elif device.type != 'cuda':
+        if device.type != 'cuda':
```

Comment:
looks like we need to adopt the changes added in #38454, to check `if device.type == 'cpu':` instead?